### PR TITLE
Add missing types for aliases querystring.encode and querystring.decode to v12 & v11

### PR DESCRIPTION
**Pull request by methodbox**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
v11.x:  https://nodejs.org/docs/latest-v11.x/api/querystring.html#querystring_querystring_encode
v12.x:  https://nodejs.org/api/querystring.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/querystring.d.ts [hunk 1]
```diff
@@ -19,6 +19,14 @@ declare module "querystring" {
 
     function stringify(obj?: ParsedUrlQueryInput, sep?: string, eq?: string, options?: StringifyOptions): string;
     function parse(str: string, sep?: string, eq?: string, options?: ParseOptions): ParsedUrlQuery;
+    /**
+     * The querystring.encode() function is an alias for querystring.stringify().
+     */
+    function encode(obj?: ParsedUrlQueryInput, sep?: string, eq?: string, options?: StringifyOptions): string;
+    /**
+     * The querystring.encode() function is an alias for querystring.stringify().
```
Comment (Flarna) [inline types/node/querystring.d.ts]: The comment is wrong here.

Comment (methodbox) [reply types/node/querystring.d.ts]: Good call. Bad copy paste on my part :(. I'll approve your correction. Thanks for catching that!

**Review by methodbox — COMMENTED** [review]
All good.

File: types/node/querystring.d.ts (hunk 1 above)
Comment (methodbox) [reply types/node/querystring.d.ts]: Sorry I got confused here. I'll push a correction then.

Comment (methodbox) [reply types/node/querystring.d.ts]: Actually, no I'm confused where you're seeing this. I'm looking at my most recent commit and I don't see this:

https://github.com/methodbox/DefinitelyTyped/commit/e4f53b61d6ffffb8a2aadd86fd3e97ad30075cd1

Comment (Flarna) [reply types/node/querystring.d.ts]: I look into the diff directly on GitHub. the file in v11 folder looks good but that one in node folder has the copy paste error.

Comment (methodbox) [reply types/node/querystring.d.ts]: Okay, I’ve resolved it - seems like my browser was caching it or something :).

File: types/node/querystring.d.ts [hunk 2]
```diff
@@ -19,6 +19,14 @@ declare module "querystring" {
 
     function stringify(obj?: ParsedUrlQueryInput, sep?: string, eq?: string, options?: StringifyOptions): string;
     function parse(str: string, sep?: string, eq?: string, options?: ParseOptions): ParsedUrlQuery;
+    /**
+     * The querystring.encode() function is an alias for querystring.stringify().
+     */
+    function encode(obj?: ParsedUrlQueryInput, sep?: string, eq?: string, options?: StringifyOptions): string;
```
Comment (islishude) [inline types/node/querystring.d.ts]: I think that the `obj` should not a optional,If ignore it,`encode` always returns an empty string.https://github.com/nodejs/node/blob/master/lib/querystring.js#L160-L197.

**Review by methodbox — COMMENTED** [review]
@islishude Can you look at this please? I understand your comment, but I think in this case it's incorrect.

**Review by islishude — APPROVED** [review]

**Comment by Flarna** [comment]
I think it would be simpler to use
```
const encode: typeof stringify;
const decode: typeof parse;
```
to express that these are aliases. This way it's ensured that they stay in sync.

**Comment by methodbox** [comment]
@Flarna But `stringify` and `parse` aren’t defined as types, they’re defined as functions which take specific types as parameters and return specific types.

I can rewrite that, but then it would be necessary to do so in all types files.

I can do that, if you think that’s best?

**Comment by methodbox** [comment]
I guess I also consider the usefulness of the type definition. Using it as I have it makes it much more verbose; referencing `typeof` means the definition seen is just `typeof X` then you have to go look up what X is ... kind of missing the point of TS.

**Comment by SimonSchick** [comment]
Duplication is not the point of TS.

**Comment by methodbox** [comment]
> Duplication is not the point of TS.

That's a bit of a snarky answer that ignores _my point:_ documentation. And I agree, duplication isn't the point, but usefulness is certainly to be considered.

If the owners want it rewritten such that you have to take two steps instead of one to find the definition of a type just so we can prevent "duplication" of code (note that this is already effectively done within this same file - not my doing - just look at escape and unescape) so be it; I'll change it.

But the definition IMO is more useful when you can see it explicitly rather than saying one thing is a `typeof` another thing, then having to look that thing up.

**Comment by Flarna** [comment]
@methodbox I tried this locally and there seems to be no need to rewrite `stringify` and `parse`. Regarding docs/code completion: At least VS Code is able to resolve this indirection and shows the full info as if you use `stringify` or `parse`.

The `typeof` solution is in use at several other places, for example in global.d.ts.

I will not block this as it is but would be nice if you could give this a try in your environment. If it doesn't work nice there keep it as it is.

**Comment by methodbox** [comment]
In my VS Code I didn’t see that but I’ll give it another look as I’m not 100% sure I tested that.

I can put in a new commit with the fix. It seems most people agree this is the way to go, and at least it also gets fixed. I’m okay with that compromise.

**Review by Flarna — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]

**Pull request merged**